File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/two/Long_ScreenTwo_Bars_LastGreen.java
package com.kishlaly.ta.analyze.tasks.blocks.two;

import com.kishlaly.ta.model.BlockResultJava;
import com.kishlaly.ta.model.QuoteJava;
import com.kishlaly.ta.model.SymbolDataJava;
import com.kishlaly.ta.utils.CollectionUtilsJava;
import com.kishlaly.ta.utils.LogJava;

import java.util.List;

import static com.kishlaly.ta.model.BlockResultCodeJava.LAST_QUOTE_NOT_GREEN_SCREEN_2;
import static com.kishlaly.ta.model.BlockResultCodeJava.OK;

/**
 * Last quote is green
 */
public class Long_ScreenTwo_Bars_LastGreen implements ScreenTwoBlockJava {
    @Override
    public BlockResultJava check(SymbolDataJava screen) {
        List<QuoteJava> screen_2_Quotes = screen.quotes;
        QuoteJava quote1 = CollectionUtilsJava.getFromEnd(screen_2_Quotes, 1);

        boolean quote1Green = quote1.getClose() > quote1.getOpen();
        if (!quote1Green) {
            LogJava.recordCode(LAST_QUOTE_NOT_GREEN_SCREEN_2, screen);
            LogJava.addDebugLine("The last quote is not green on the second screen");
            return new BlockResultJava(screen.getLastQuote(), LAST_QUOTE_NOT_GREEN_SCREEN_2);
        }
        return new BlockResultJava(screen.getLastQuote(), OK);
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/blocks/two/Long_ScreenTwo_Bars_LastGreen.kt
package com.kishlaly.ta.analyze.tasks.blocks.two

import com.kishlaly.ta.analyze.tasks.blocks.two.ScreenTwoBlock
import com.kishlaly.ta.model.BlockResult
import com.kishlaly.ta.model.BlockResultCode
import com.kishlaly.ta.model.SymbolData
import com.kishlaly.ta.utils.Log

/**
 * Last quote is green
 */
class Long_ScreenTwo_Bars_LastGreen : ScreenTwoBlock {
    override fun check(screen: SymbolData): BlockResult {
        val lastQuote = screen.lastQuote

        val green = lastQuote.close > lastQuote.open
        if (!green) {
            Log.recordCode(BlockResultCode.LAST_QUOTE_NOT_GREEN_SCREEN_2, screen)
            Log.addDebugLine("The last quote is not green on the second screen")
            return BlockResult(lastQuote, BlockResultCode.LAST_QUOTE_NOT_GREEN_SCREEN_2)
        }
        return BlockResult(lastQuote, BlockResultCode.OK)
    }
}